### PR TITLE
[extractor/pornhub][fix] fix changed ageverification cookie

### LINKED ISSUE #4299: [PornHub] Unable to extract encoded url
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.06.29** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

[PornHub] Unable to download JSON metadata: HTTP Error 403: Forbidden
How to fix it?

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
F:\downloader\pornhub>ytdl-patched https://www.pornhub.com/model/katekuray -F --verbose
[debug] Command-line config: ['https://www.pornhub.com/model/katekuray', '-F', '--verbose']
[debug] User config "D:\OS Files\Users\X\AppData\Roaming\yt-dlp\config": ['--continue', '--http-chunk-size', '1M', '--buffer-size', '16K', '--sub-lang', 'english', '--embed-metadata', '--abort-on-unavailable-fragment', '-o', '%(uploader)s\\%(title)s [%(id)s].%(ext)s', '-f', 'bestvideo[ext=mp4]+bestaudio[ext=m4a]', '--ffmpeg-location', 'D:\\OS Files\\Program Files\\ffmpeg\\bin', '--no-lock']
[debug] Encodings: locale cp65001, fs utf-8, pref cp65001, out utf-8, error utf-8, screen utf-8
[debug] ytdl-patched version 2022.06.29
[debug] ** The command you are running is not yt-dlp.
[debug] ** Please make bug reports at https://github.com/ytdl-patched/ytdl-patched/issues/new/choose instead.
[debug] Lazy loading extractors is disabled
[debug] Python 3.10.4 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] Checking exe version: "D:\OS Files\Program Files\ffmpeg\bin\ffmpeg" -bsfs
[debug] Checking exe version: "D:\OS Files\Program Files\ffmpeg\bin\ffprobe" -bsfs
[debug] exe versions: ffmpeg 5.0.1-essentials_build-www.gyan.dev (setts), ffprobe 5.0.1-essentials_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.14.1, WebSocketsWrapper-None, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.2
[debug] Proxy map: {}
[debug] [PornHubUser] Extracting URL: https://www.pornhub.com/model/katekuray
[debug] [PornHubPagedVideoList] Extracting URL: https://www.pornhub.com/model/katekuray/videos
[download] Downloading playlist: model/katekuray/videos
[PornHubPagedVideoList] model/katekuray/videos: Downloading page 1
[PornHubPagedVideoList] [debug] Fetching webpage from https://www.pornhub.com/model/katekuray/videos
[PornHubPagedVideoList] model/katekuray/videos: Downloading page 2
[PornHubPagedVideoList] [debug] Fetching webpage from https://www.pornhub.com/model/katekuray/videos
[PornHubPagedVideoList] Playlist model/katekuray/videos: Downloading 107 videos of 107
[download] Downloading video 11 of 107
[PornHub] ph5f71889303342: Downloading pc webpage
[PornHub] ph5f71889303342: Downloading m3u8 information
[PornHub] ph5f71889303342: Downloading m3u8 information
[PornHub] ph5f71889303342: Downloading m3u8 information
[PornHub] ph5f71889303342: Downloading m3u8 information
[PornHub] ph5f71889303342: Downloading JSON metadata
WARNING: [PornHub] Unable to download JSON metadata: HTTP Error 403: Forbidden
[info] Available formats for ph5f71889303342:
ID         EXT RESOLUTION FPS │  FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────────
hls-509-0  mp4 426x240     59 │ ~ 4.48MiB  509k m3u8  │ avc1.64001e  509k mp4a.40.2  0k
hls-509-1  mp4 426x240     59 │ ~ 4.48MiB  509k m3u8  │ avc1.64001e  509k mp4a.40.2  0k
hls-902-0  mp4 854x480     59 │ ~ 7.93MiB  902k m3u8  │ avc1.64001f  902k mp4a.40.2  0k
hls-902-1  mp4 854x480     59 │ ~ 7.93MiB  902k m3u8  │ avc1.64001f  902k mp4a.40.2  0k
hls-1614-0 mp4 1280x720    59 │ ~14.19MiB 1614k m3u8  │ avc1.640020 1614k mp4a.40.2  0k
hls-1614-1 mp4 1280x720    59 │ ~14.19MiB 1614k m3u8  │ avc1.640020 1614k mp4a.40.2  0k
hls-2769   mp4 1920x1080   59 │ ~24.34MiB 2769k m3u8  │ avc1.64002a 2769k mp4a.40.2  0k
```

### PR DESCRIPTION
**Pull request by Schmoaaaaah**
**IMPORTANT**: PRs without the template will be CLOSED

### [extractor/pornhub][fix] adds fix for a changed ageverification cookie

It adds two additional cookies to the extractor that were recently introduced by the site and prohibit the access to download the metadata.

The fix is from https://github.com/ytdl-org/youtube-dl/pull/31916

Closes #4299

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [X] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [X] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/pornhub.py [hunk 1]
```diff
@@ -269,6 +269,8 @@ def _real_extract(self, url):
         self._login(host)
 
         self._set_cookie(host, 'age_verified', '1')
+        self._set_cookie(host, 'accessAgeDisclaimerPH', '1')
+        self._set_cookie('thumbzilla.com', 'accessAgeDisclaimerTZ', '1')
```
Comment (tinyoverflow) [inline yt_dlp/extractor/pornhub.py:273]: Why do you set a cookie for a third party site? This shouldn't be required.

Comment (bashonly) [reply yt_dlp/extractor/pornhub.py:273]: the extractor is used for both sites

```py
class PornHubIE(PornHubBaseIE):
    IE_DESC = 'PornHub and Thumbzilla'
```

Comment (tinyoverflow) [reply yt_dlp/extractor/pornhub.py:273]: Oh, didn't saw that. I'm sorry.

**Comment by Menard01** [comment]
It seems to be broken again

File: yt_dlp/extractor/pornhub.py (hunk 1 above)
Comment (bashonly) [reply yt_dlp/extractor/pornhub.py:273]: after looking more closely at the code, I think you may be right that the thumbzilla cookie is unnecessary. The extractor seems to match thumbzilla URLs only. Everything is extracted from PH

**Comment by bashonly** [comment]
see https://github.com/yt-dlp/yt-dlp/issues/4299#issuecomment-1499715987

**Comment by bashonly** [comment]
based on [comments in the linked issue](https://github.com/yt-dlp/yt-dlp/issues/4299#issuecomment-1501182785) and [these review comments](https://github.com/yt-dlp/yt-dlp/pull/6685#discussion_r1160318580), I think something like this needs to be done:
```diff
diff --git a/yt_dlp/extractor/pornhub.py b/yt_dlp/extractor/pornhub.py
index fe26f52a9..d830919f1 100644
--- a/yt_dlp/extractor/pornhub.py
+++ b/yt_dlp/extractor/pornhub.py
@@ -58,6 +58,10 @@ def dl(*args, **kwargs):
     def _real_initialize(self):
         self._logged_in = False
 
+    def _set_age_cookies(self, host):
+        self._set_cookie(host, 'age_verified', '1')
+        self._set_cookie(host, 'accessAgeDisclaimerPH', '1')
+
     def _login(self, host):
         if self._logged_in:
             return
@@ -267,10 +271,7 @@ def _real_extract(self, url):
         video_id = mobj.group('id')
 
         self._login(host)
-
-        self._set_cookie(host, 'age_verified', '1')
-        self._set_cookie(host, 'accessAgeDisclaimerPH', '1')
-        self._set_cookie('thumbzilla.com', 'accessAgeDisclaimerTZ', '1')
+        self._set_age_cookies(host)
 
         def dl_webpage(platform):
             self._set_cookie(host, 'platform', platform)
@@ -571,6 +572,7 @@ def _real_extract(self, url):
         mobj = self._match_valid_url(url)
         user_id = mobj.group('id')
         videos_url = '%s/videos' % mobj.group('url')
+        self._set_age_cookies(mobj.group('host'))
         page = self._extract_page(url)
         if page:
             videos_url = update_url_query(videos_url, {'page': page})
@@ -635,6 +637,7 @@ def _real_extract(self, url):
         item_id = mobj.group('id')
 
         self._login(host)
+        self._set_age_cookies(host)
 
         return self.playlist_result(self._entries(url, host, item_id), item_id)
 
@@ -814,8 +817,6 @@ def _real_extract(self, url):
         item_id = mobj.group('id')
 
         self._login(host)
-
-        self._set_cookie(host, 'accessAgeDisclaimerPH', '1')
-        self._set_cookie('thumbzilla.com', 'accessAgeDisclaimerTZ', '1')
+        self._set_age_cookies(host)
 
         return self.playlist_result(self._entries(mobj.group('url'), host, item_id), item_id)
```

**Comment by Schmoaaaaah** [comment]
@bashonly Yea i thought about that too and tested adding the cookies to all the extract functions, but only playlist and video dont throw an error without it. The other dont throw an error regardless but also dont work with either the cookie set or not set🥴. I assume the problem with that lies some were else but i am not sure.

**Comment by bashonly** [comment]
The above comment is not helpful. If you are commenting to say it doesn't work for you, at least post a verbose log and an explanation of how you applied the patch.

**Comment by thoastbrot** [comment]
works on my machine - I built yt-dlp using the source repo of this PR (not a patch file). Using the last official release gets me the error @tantined-lotus has posted.

```
./yt-dlp "https://www.pornhub.com/view_video.php?viewkey=5ee72898128dd"  --verbose
[debug] Command-line config: ['https://www.pornhub.com/view_video.php?viewkey=5ee72898128dd', '--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (zip)
[debug] Python 3.9.2 (CPython x86_64 64bit) - Linux-5.10.0-21-amd64-x86_64-with-glibc2.31 (OpenSSL 1.1.1n  15 Mar 2022, glibc 2.31)
[debug] exe versions: ffmpeg 4.3.5-0, ffprobe 4.3.5-0
[debug] Optional libraries: brotli-1.0.9, certifi-2020.06.20, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1791 extractors
[PornHub] Extracting URL: https://www.pornhub.com/view_video.php?viewkey=5ee72898128dd
[PornHub] 5ee72898128dd: Downloading pc webpage
[PornHub] 5ee72898128dd: Downloading m3u8 information
[PornHub] 5ee72898128dd: Downloading m3u8 information
[PornHub] 5ee72898128dd: Downloading m3u8 information
[PornHub] 5ee72898128dd: Downloading m3u8 information
[PornHub] 5ee72898128dd: Downloading JSON metadata
```

**Comment by bashonly** [comment]
@tantined-lotus it doesn't look like you are doing anything wrong. Since the cookie issue is tied to location (I still am not affected), it may be that a different cookie is required for requests originating from your region/country? Could be helpful if you want to share what that is. Or you could do some browser dev tools investigation to see what cookies PH sets when you are browsing the website.

**Comment by Schmoaaaaah** [comment]
@tantined-lotus I tried `pip install git+https://github.com/Schmoaaaaah/yt-dlp.git` and it works for me.
Might you add from what Country you trying to access. Then i can try with a vpn.

**Comment by Schmoaaaaah** [comment]
I just tried with a VPN to Netherlands and it worked fine.
I used this Dockerfile:
```Dockerfile
FROM python

RUN apt update
RUN apt install -y ffmpeg

RUN pip install git+https://github.com/Schmoaaaaah/yt-dlp.git

CMD ["yt-dlp", "--verbose", "https://www.pornhub.com/view_video.php?viewkey=5ee72898128dd"]
```

**Comment by Schmoaaaaah** [comment]
Where you able to run it through a vpn?
My best guess at the moment is something is diffrent in your web connection, some dns filtering that is interferring or similar.

**Comment by bashonly** [comment]
> you could do some browser dev tools investigation to see what cookies PH sets when you are browsing the website.

@tantined-lotus if you are able to do this, that would be helpful, since no one else is able to reproduce the issue you are having with the patch

File: yt_dlp/extractor/pornhub.py [hunk 2]
```diff
@@ -58,6 +58,10 @@ def dl(*args, **kwargs):
     def _real_initialize(self):
         self._logged_in = False
 
+    def _set_age_cookies(self, host):
+        self._set_cookie(host, 'age_verified', '1')
+        self._set_cookie(host, 'accessAgeDisclaimerPH', '1')
```
Comment (bashonly) [inline yt_dlp/extractor/pornhub.py:63]:
```suggestion
        self._set_cookie(host, 'accessAgeDisclaimerPH', '1')
        self._set_cookie(host, 'accessPH', '1')
```

**Comment by bashonly** [comment]
@tantined-lotus can you try it now with the change just applied?

**Comment by pukkandan** [comment]
(Note to maintainers: Despite the small size of the patch, this seems to have multiple authors, Make sure to add `Authored by:` correctly)

**Comment by Schmoaaaaah** [comment]
@bashonly works perfectly for me. I am a bit Lost to what to do with the `Authored by:`

**Comment by bashonly** [comment]
@Schmoaaaaah I'll take care of it

**Comment by thoastbrot** [comment]
No new release since merge:
https://github.com/yt-dlp/yt-dlp/releases

Patience, @tiritibambix

**Comment by bashonly** [comment]
`yt-dlp --update-to nightly`

**Comment by tiritibambix** [comment]
> `yt-dlp --update-to nightly`

I'm using youtubedl-material. Not sure how to update within the container though

**Comment by tinyoverflow** [comment]
> I'm using youtubedl-material. Not sure how to update within the container though

@tiritibambix You should ask that on the corresponding project page, as youtubedl-material is not maintained in this repository.

**Comment by tiritibambix** [comment]
Will do, thank you :)

**Pull request merged**